framework/src/tinyalsa/tinyalsa.c

```diff
@@ -1120,16 +1136,12 @@ int pcm_stop(struct pcm *pcm)
 	unsigned int size;
 	int prio;
 	struct timespec st_time;
-	while (pcm->buf_ptr > 0) {
+	while (1) {
```

Comment:
why you rollback code here? Is there any reason?

Reply:
I have rolled back a few changes related to buf_ptr. 
When i introduced buf_ptr variable, it was meant to keep track of the buffers to be enqueued only at the start of playback or record. For example, if we have 4 buffers, buf_ptr will be incremented in writei or readi when these 4 buffers are enqueued the first time only. It is not supposed to be used to track the enque-deque of buffers later on. Using it in this way is causing a problem at another point. So i reverted it. I think this implementation works well enough.

Reply:
But, I think i will change this implementation slightly.

Reply:
Got it.